src/vs/workbench/services/workingCopy/common/workingCopyService.ts

```diff
@@ -138,6 +139,11 @@ export class WorkingCopyService extends Disposable implements IWorkingCopyServic
 	registerWorkingCopy(workingCopy: IWorkingCopy): IDisposable {
 		const disposables = new DisposableStore();
 
+		// {{SQL CARBON EDIT}} @chlafreniere need to block working copies of notebook editors from being tracked
+		if (workingCopy.resource.path.includes('notebook-editor-') && workingCopy.resource.scheme === Schemas.untitled) {
```

Comment:
This doesn't look too ugly, but like I said I think the correct solution is to create a custom input and model for those code cells so we have better control over what is talking to those cells.

Reply:
Ack